internals-js/src/specs/cacheTagSpec.ts

```diff
@@ -0,0 +1,75 @@
+import { DirectiveLocation, GraphQLError } from 'graphql';
+import {
+  CorePurpose,
+  FeatureDefinition,
+  FeatureDefinitions,
+  FeatureUrl,
+  FeatureVersion,
+} from './coreSpec';
+import { Schema } from '../definitions';
+import { registerKnownFeature } from '../knownCoreFeatures';
+import { createDirectiveSpecification } from '../directiveAndTypeSpecification';
+
+export const cacheTagIdentity = 'https://specs.apollo.dev/cacheTag';
+
+const CACHE_KEY = 'cacheTag';
+const FORMAT = 'format';
+
+export class CacheTagSpecDefinition extends FeatureDefinition {
+  constructor(
+    version: FeatureVersion,
+    readonly minimumFederationVersion: FeatureVersion,
+  ) {
+    super(
+      new FeatureUrl(cacheTagIdentity, CACHE_KEY, version),
+      minimumFederationVersion,
+    );
+
+    this.registerDirective(
+      createDirectiveSpecification({
+        name: CACHE_KEY,
+        locations: [
+          DirectiveLocation.FIELD_DEFINITION,
+          DirectiveLocation.OBJECT,
+        ],
+        repeatable: true,
+        // We "compose" these directives using the  `@join__directive` mechanism,
+        // so they do not need to be composed in the way passing `composes: true`
+        // here implies.
+        composes: false,
+      }),
+    );
+  }
+
+  addElementsToSchema(schema: Schema): GraphQLError[] {
+    /*
+      directive @cacheTag(
+        format: String
+      ) repeatable on FIELD_DEFINITION
+        | OBJECT
+    */
+    const cacheTag = this.addDirective(schema, CACHE_KEY).addLocations(
+      DirectiveLocation.FIELD_DEFINITION,
+      DirectiveLocation.OBJECT,
+    );
+    cacheTag.repeatable = true;
+
+    cacheTag.addArgument(FORMAT, schema.stringType());
```

Comment:
should this be non-null? i.e. what would be the use case for `@cacheTag` without any value?